docs/src/considerations.md

```diff
@@ -15,3 +15,82 @@ Here's a collection of references for developers who are interested:
+    end)
+end
+```
+
+and
+
+```julia
+function basetype(gt::OGRwkbGeometryType)::OGRwkbGeometryType
+    wkbGeomType = convert(GDAL.OGRwkbGeometryType, gt)
+    wkbGeomType &= (~0x80000000) # Remove 2.5D flag.
+    wkbGeomType %= 1000 # Normalize Z, M, and ZM types.
+    return GDAL.OGRwkbGeometryType(wkbGeomType)
+end
+```
+
+However, the use of CEnum.jl allows for multiple enums to have the same underlying value, resulting in unintuitive behavior if they are used as keys in a dictionary. For example, in the following code:
+
+```julia
+julia> Dict(GDAL.GCI_YCbCr_CrBand => "a", GDAL.GCI_Max => "b")
+Dict{GDAL.GDALColorInterp, String} with 1 entry:
+  GCI_YCbCr_CrBand => "b"
+```
+
+the entry for `GDAL.GCI_YCbCr_CrBand => "a"` got overwritten by `GDAL.GCI_Max => "b"` because both `GDAL.GCI_YCbCr_CrBand` and `GDAL.GCI_Max` corresponded to the same value.
+
+To avoid such forms of behavior, this package uses [`Base.Enums`](https://docs.julialang.org/en/v1/base/base/#Base.Enums.Enum) instead, so the above example would result in the following behavior:
+
+
+```julia
+julia> Dict(ArchGDAL.GCI_YCbCr_CrBand => "a", ArchGDAL.GCI_Max => "b")
+Dict{ArchGDAL.GDALColorInterp, String} with 2 entries:
+  GCI_YCbCr_CrBand => "a"
+  GCI_Max          => "b"
+```
+
+To maintain parity with GDAL behavior, ArchGDAL.jl provides conversion methods to map from the enums in ArchGDAL to the corresponding cenums from GDAL.jl when calling the corresponding GDAL functions.
+
+## Colors
+
+Rather than encouraging [operations on colortables](https://gdal.org/python/osgeo.gdal.ColorTable-class.html) (with very limited functionality from GDAL), users are better served by arrays of [ColorTypes](https://github.com/JuliaGraphics/ColorTypes.jl) using [Colors.jl](https://github.com/JuliaGraphics/Colors.jl), [for example](http://juliagraphics.github.io/Colors.jl/stable/colormapsandcolorscales/#Generating-a-range-of-colors)
+
+```julia
+range(startcolor, stop=endcolor, length=15)
+```
+
+instead of
+
+```julia
+createcolorramp!(colortable, startindex, startcolor, startindex+15, endcolor)
+```
+
+## Images
+
+To differentiate 2d arrays of colors from 3d arrays with band as the third dimension:
+
+* For 2D arrays (a single rasterband), if they have a color interpretation, we use the color interpretation. If they don't have a color interpretation, ArchGDAL.jl defaults to "Grey".
+* For >2D arrays (multiple rasterbands), if they have a palette interpretation (or combination of color interpretations) that resolves to a valid colortype, ArchGDAL.jl uses the palette interpretation. If they don't have a palette interpretation, we throw an error.
+
+In general, `read()` will return `Array{UInt8}`, and `imread()` will return `Array{<:Colorant}`.
+
+## Tables.jl Interface
+
+The interface is implemented in [`src/tables.jl`](https://github.com/yeesian/ArchGDAL.jl/blob/master/src/tables.jl). The current API from GDAL makes it row-based in the conventions of Tables.jl. Therefore,
```

Comment:
Note that this interface is for `vector` stuff only. Unlike the colors, which is `raster` stuff.

Reply:
Thank you! https://github.com/yeesian/ArchGDAL.jl/pull/179/commits/b53179d0d133d510a3427bc4cefb4a4b9003bbd9